packages/calcite-components/src/components/tile-group/tile-group.tsx

```diff
@@ -52,16 +102,79 @@ export class TileGroup implements InteractiveComponent {
 
   @Element() el: HTMLCalciteTileGroupElement;
 
+  private items: HTMLCalciteTileElement[] = [];
+
+  private slotEl: HTMLSlotElement;
+
   //--------------------------------------------------------------------------
   //
   //  Private Methods
   //
   //--------------------------------------------------------------------------
 
+  private getSlottedTiles = (): HTMLCalciteTileElement[] => {
+    return this.slotEl
+      ?.assignedElements({ flatten: true })
+      .filter((el) => el?.matches("calcite-tile")) as HTMLCalciteTileElement[];
+  };
+
+  private mutationObserver = createObserver("mutation", () => this.updateTiles());
+
+  private selectItem = (item: HTMLCalciteTileElement): void => {
+    if (!item) {
+      return;
+    }
+    this.items?.forEach((el) => {
+      const matchingEl = item === el;
+      switch (this.selectionMode) {
+        case "multiple":
+          if (matchingEl) {
+            el.selected = !el.selected;
+          }
+          break;
+
+        case "single":
+          el.selected = matchingEl ? !el.selected : false;
```

Comment:
Suggestion: `el.selected = matchingEl && !el.selected;`. This is minor, so I'll leave this up to you.